test/metabase/util/queue_test.clj

```diff
@@ -97,22 +98,131 @@
     ;; queue an outlier
     (queue/put-with-delay! q (msg-delay 10) 10)
     (let [started-roughly (u/start-timer)
-          since-start     #(u/since-ms started-roughly)
-          time-until-nth  #(max 0 (+ buffer (- (msg-delay %) (since-start))))
-          additional-wait (+ extra-delay buffer)]
-      (testing "Initially none of the messages are ready"
-        (is (nil? (queue/take-delayed-batch! q batch-size))))
-      (testing "Polling for a short time will also return before any messages are ready"
-        (is (nil? (queue/take-delayed-batch! q batch-size (quot first-delay 2) additional-wait))))
-      (testing "Waiting a bit longer, we will retrieve a batch"
-        (is (= [0 1 2] (queue/take-delayed-batch! q batch-size (time-until-nth 0) additional-wait))))
-      ;; Wait until all items have matured
-      (Thread/sleep ^long (time-until-nth n))
-      (testing "Some time later we can read an additional batch of messages without any polling delay"
-        (is (= [3 4] (queue/take-delayed-batch! q batch-size))))
-      (testing "But the outlier is not yet ready"
-        (is (nil? (queue/take-delayed-batch! q batch-size))))
-      (testing "Eventually the outlier is ready"
-        (is (= [10] (queue/take-delayed-batch! q batch-size (time-until-nth 10) additional-wait))))
-      (testing "Afterwards the queue is empty"
-        (is (nil? (queue/take-delayed-batch! q batch-size)))))))
+          since-start #(u/since-ms started-roughly)
+          time-until-nth #(max 0 (+ buffer (- (msg-delay %) (since-start))))]
+      (binding [queue/*take-batch-wait-ms* (time-until-nth 10)]
+        (testing "Take-batch will wait for the first message to be ready"
```

Comment:
I guess we were over testing the delay aspect before? Feels a bit weird cutting so many assertions though 😄

Reply:
I went back to compare with the original version after bringing back the max-next-ms logic and tried to make sure we're testing the same stuff as before, just with a little cleanup that made it make more sense to me. If any tests seem missing still, let me know